core/extstate/statedb_multicoin_test.go

```diff
@@ -1,133 +1,147 @@
 // Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE for licensing terms.
 
-package state
+package extstate
 
 import (
 	"math/big"
 	"testing"
 
+	"github.com/ava-labs/coreth/core/state"
 	"github.com/ava-labs/coreth/core/state/snapshot"
 	"github.com/ava-labs/coreth/plugin/evm/customtypes"
 	"github.com/ava-labs/libevm/common"
 	"github.com/ava-labs/libevm/core/rawdb"
+	"github.com/ava-labs/libevm/core/types"
 	"github.com/ava-labs/libevm/crypto"
 	"github.com/ava-labs/libevm/libevm/stateconf"
 	"github.com/holiman/uint256"
 	"github.com/stretchr/testify/require"
 )
 
 func TestMultiCoinOperations(t *testing.T) {
-	s := newStateEnv()
+	rawdb := rawdb.NewMemoryDatabase()
+	db := state.NewDatabase(rawdb)
+	stateDB, err := state.New(types.EmptyRootHash, db, nil)
+	require.NoError(t, err, "creating empty statedb")
+
 	addr := common.Address{1}
 	assetID := common.Hash{2}
 
-	root, err := s.state.Commit(0, false)
-	require.NoError(t, err, "committing state")
-	s.state, err = New(root, s.state.db, s.state.snaps)
-	require.NoError(t, err, "creating statedb")
-
-	s.state.AddBalance(addr, new(uint256.Int))
+	stateDB.AddBalance(addr, new(uint256.Int))
 
-	balance := s.state.GetBalanceMultiCoin(addr, assetID)
+	state := New(stateDB)
```

Comment:
nit: These overlapping names is hard
```suggestion
	wrappedStateDB := New(stateDB)
```